### ISSUE TITLE
Website: perf tweaks

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#484]
- [ ] gzip
- [ ] explicit width and height on images (build step)
- [ ] cache images more aggressively than index.html
- [ ] spritesheet icons?
- [ ] minify CSS?
- [ ] resize some images (e.g. codespaces.png)

**Comment by Youness-Asserare** [comment]
Hey can this issue be assigned to me?

**Comment by Youness-Asserare** [comment]
I made improvements , only the following tasks are left:
 gzip
spritesheet icons (SVG <use>

**Comment by Youness-Asserare** [comment]
i have been trying to solve the gzip part but couldn't ! do you have any helpful ideas to help me solve it ?

**Comment by strager** [comment]
> i have been trying to solve the gzip part but couldn't !

It looks like gzip encoding is already happening for some files, but not all:

<img width="905" alt="Screen Shot 2022-01-27 at 10 15 16 AM" src="https://user-images.githubusercontent.com/48666/151419286-74207bed-102d-4ba3-bfbf-f7b474c1ac28.png">

Perhaps we just need to add .svg to a whitelist of gzippable extensions in the Apache config. (.gif, .png, and .webp don't need gzipping.)

**Comment by strager** [comment]
@Youness-Asserare Can you share your current progress in a pull request?

**Comment by Youness-Asserare** [comment]
sorry but I lost all of the progress i have made. I will try solving a
different issue.

**Comment by Youness-Asserare** [comment]
by spritesheet icons, you mean this https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Images/Implementing_image_sprites_in_CSS ?

**Comment by strager** [comment]
> by spritesheet icons, you mean this https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Images/Implementing_image_sprites_in_CSS ?

No. I'm talking about SVG's `<symbol>` and `<use>`. I think this article showcases what we want: https://css-tricks.com/svg-symbol-good-choice-icons/
